### PR TITLE
Homework #4

### PR DESCRIPTION
**Pull request by zan-od**

### CODE REVIEW

File: src/main/java/hillel/spring/doctor/controller/DoctorController.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        return ResponseEntity.created(new URI("/doctors/" + doctor.getId())).build();
+    }
+
+    @PutMapping("/doctors/{id}")
+    public ResponseEntity<?> update(@RequestBody Doctor doctor,
+                                    @PathVariable("id") Integer id) {
+
+        Assert.notNull(id, "Path variable {id} not specified");
+        Assert.notNull(doctor.getId(), "Doctor's id not specified");
```
Comment (mfarsikov) [inline src/main/java/hillel/spring/doctor/controller/DoctorController.java]: this leads to 500 error, but it should be 400 instead

File: src/main/java/hillel/spring/doctor/repository/DoctorRepository.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.stream.Collectors;
+
+@Repository
+public class DoctorRepository {
+
+    private final List<Doctor> doctors = Collections.synchronizedList(new ArrayList<>());
+    private final AtomicInteger lastId = new AtomicInteger(0);
+
+    public List<Doctor> list() {
+        return Collections.unmodifiableList(doctors);
```
Comment (mfarsikov) [inline src/main/java/hillel/spring/doctor/repository/DoctorRepository.java]: This is called "defensive programming". It makes sense when your product is a library. But usually, the only users of this code are you and your team-mates. So Defensive programming is often considered as overkill.

**Review by mfarsikov — APPROVED** [review]

File: src/main/java/hillel/spring/doctor/controller/DoctorController.java (hunk 1 above)
Comment (zan-od) [reply src/main/java/hillel/spring/doctor/controller/DoctorController.java]: fixed

File: src/main/java/hillel/spring/doctor/repository/DoctorRepository.java (hunk 2 above)
Comment (zan-od) [reply src/main/java/hillel/spring/doctor/repository/DoctorRepository.java]: I see, thank you. Fixed.

**Pull request merged**